seacrowd/sea_datasets/basahacorpus/basahacorpus.py

```diff
@@ -0,0 +1,217 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+BasahaCorpus contains short stories in four Central Philippine languages \
+    (Minasbate, Rinconada, Kinaray-a, and Hiligaynon) for low-resource \
+    readability assessment. Each dataset per language contains stories \
+    distributed over the first three grade levels (L1, L2, and L3) in \
+    the Philippine education context. The grade levels of the dataset \
+    have been provided by an expert from Let's Read Asia.
+"""
+
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+import pandas as pd
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Tasks, Licenses
+
+_CITATION = """\
+@misc{imperial2023basahacorpus,
+      title={BasahaCorpus: An Expanded Linguistic Resource for Readability Assessment in Central Philippine Languages}, 
+      author={Joseph Marvin Imperial and Ekaterina Kochmar},
+      year={2023},
+      eprint={2310.11584},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+}
+"""
+
+_DATASETNAME = "basaha_corpus"
+
+_DESCRIPTION = """
+BasahaCorpus contains short stories in four Central Philippine languages \
+    (Minasbate, Rinconada, Kinaray-a, and Hiligaynon) for low-resource \
+    readability assessment. Each dataset per language contains stories \
+    distributed over the first three grade levels (L1, L2, and L3) in \
+    the Philippine education context. The grade levels of the dataset \
+    have been provided by an expert from Let's Read Asia.
+"""
+_HOMEPAGE = "https://github.com/imperialite/BasahaCorpus-HierarchicalCrosslingualARA"
+
+_LANGUAGES = [
+    "msb",
+    "rin",
+    "kar",
+    "hil",
```

Comment:
Some of the language codes differ from the issue card, which includes "bto", "krj", "hil", "msb". On the other hand, in the code we have "kar" and "rin" which are not listed. There might be difference in the language code in this dataloader implementation, the original dataset, and the issue card. 

I think "bto" might refer to "bik" in the original dataset, and "krj" refers to "kar" in the original dataset.
Also, the language codes listed here seems to follow the original paper. 

We need to either add the "bto", "krj" and remove "rin" or follow the original paper's (or original dataset) language.

Need clarification on the language codes @SamuelCahyawijaya @holylovenia

Reply:
> Figure 1. The central subgroup of the Philippine language family tree highlighting the origins of the target languages Hiligaynon, Minasbate, Karay-a, and Rinconada (both underlined and marked with *). Tagalog, Bikol, and Cebuano are also underlined as they are part of further experiments.

After reading the paper [here](https://arxiv.org/abs/2310.11584), and see Figure 1, it seems that the target languages are only four: Hiligaynon (`hil`), Minasbate (`msb`), Karay-a (`kar`), and Rinconada (`rin`).

Tagalog, Bikol, and Cebuano are not the (main) target language here. From their Github homepage, in [this folder](https://github.com/imperialite/BasahaCorpus-HierarchicalCrosslingualARA/tree/main/data/features). We can see that there are exist Tagalog (`tag`), Bikol (`bik`), and Cebuano (`ceb`).

But in my consideration, I don't take those language because it's not the main target language on this task.

However, I still waiting the @SamuelCahyawijaya @holylovenia response for this matter. I won't resolve this conversation until the response.

Thanks for the review though @jen-santoso!